emcc.py

```diff
@@ -1217,7 +1217,10 @@ def check(input_file):
         newargs.append('-D__EMSCRIPTEN_PTHREADS__=1')
         shared.Settings.FORCE_FILESYSTEM = 1 # proxying of utime requires the filesystem
       else:
-        js_libraries.append(shared.path_from_root('src', 'library_pthread_stub.js'))
+        if final_suffix in JS_CONTAINING_SUFFIXES:
```

Comment:
this will work, but it means every regular (non-pthreads) compilation will have an extra C input file?

instead, how about

1. Converting those js methods to asm.js (the js versions do actual work, but your c versions are just fake, so that seems easy)
2. Add that C file as a library, or perhaps make it part of non-multithreaded libc.

?